UnityProject/Storybook/Assets/Code/Scripts/Game/World/MapManager.cs

```diff
@@ -154,29 +175,29 @@ private RoomObject _determineDoorPlacement(Location gridPosition, RoomObject the
         RoomData currentRoomData = m_worldMapData[gridPosition.X, gridPosition.Y];
 
         // Initialize the room through door locations
```

Comment:
Door placement should be determined by the room object. The room object can extend IConstructable and MapManager can call Construct on it. The parameter it takes can be a RoomData. This will provide easier to read code and easier to test code.